src/cpp/rtps/builtin/discovery/participant/PDPServer2.cpp

```diff
@@ -248,17 +252,59 @@ void PDPServer2::initializeParticipantProxyData(
 void PDPServer2::assignRemoteEndpoints(
         ParticipantProxyData* pdata)
 {
-    (void)pdata;
-    // TODO DISCOVERY SERVER VERSION 2
+    logInfo(RTPS_PDP, "For RTPSParticipant: " << pdata->m_guid.guidPrefix);
+
+    const NetworkFactory& network = mp_RTPSParticipant->network_factory();
+    uint32_t endp = pdata->m_availableBuiltinEndpoints;
+    bool use_multicast_locators = !mp_RTPSParticipant->getAttributes().builtin.avoid_builtin_multicast ||
+            pdata->metatraffic_locators.unicast.empty();
+
+    // only SERVER and CLIENT participants will be received. All builtin must be there
+    uint32_t auxendp = endp & DISC_BUILTIN_ENDPOINT_PARTICIPANT_ANNOUNCER;
+    assert(0 != auxendp);
```

Comment:
On the one hand, assert will only get triggered in debug. On the other hand, the check should always be true, it's probably imposible to hit a situation where the sender is not a builtin